Sprint-3/todo-list/script.js

```diff
@@ -16,10 +54,32 @@ populateTodoList(todos);
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let todoInput = document.getElementById("todo-input");
+  let task = todoInput.value;
+  let todoDeadline = document.getElementById("todo-deadline");
+  let deadline = todoDeadline.value;
+
+
+  if (task.trim() !== "") {
+    todos.push({ task: task, completed: false, deadline: deadline || null });
+    populateTodoList(todos);
+    todoInput.value = "";
+    todoDeadline.value = "";
+  }
 }
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+let addTodoButton = document.getElementById("add-todo");
+addTodoButton.addEventListener("click", addNewTodo);
+
+// Advanced challenge: Write a function that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
-  // Write your code here...
+  todos = todos.filter((todo) => {
+    let listItems = Array.from(document.getElementById("todo-list").children);
+    let matchedItem = listItems.find((li) => li.textContent.includes(todo.task));
+    return matchedItem && matchedItem.style.textDecoration !== "line-through";
```

Comment:
What if multiple todo's task have the same text? For example,
```
let todos = [
  { task: "Wash the dishes", completed: false, deadline: "2024-12-31" },
  { task: "Do the shopping", completed: false, deadline: "2024-12-25" },
  { task: "Wash the dishes", completed: true, deadline: "2024-12-29" },
];
```

If you focus on updating the application state (the array `todos`), the filtering task would be much simpler.